FIX: Use mg5amcnlo-pythia8-interface for MG5aMC_PY8_interface

matthewfeickert opened the pull request:
Partially addresses Issue #7 (requires https://github.com/conda-forge/mg5amcnlo-pythia8-interface-feedstock/pull/4 also)

**FIX: Use mg5amcnlo-pythia8-interface for MG5aMC_PY8_interface**

* Add `mg5amcnlo-pythia8-interface` as a `run` requirement, which supplies the Pythia8 interface program and defines `pythia8` `run` requirements.
   - Skip `mg5amcnlo-pythia8-interface` for `osx-64` given `PY8_parallelization` splitting bug.
   - c.f. https://github.com/conda-forge/mg5amcnlo-pythia8-interface-feedstock/issues/2
* Remove build information for MG5aMC_PY8_interface and remove file checks.
* Bump build number.

**FIX: Use target platform compiler names in mg5_configuration.txt**

* Use `CXX_FOR_BUILD` and `FF_FOR_BUILD` to get the correct compilers that for the runtime target platform (the host platform). The '_FOR_BUILD' compilers give the build platform compilers, but with the correct name prefaced by the `BUILD` triplet (e.g. `x86_64-conda-linux-gnu`).
   - c.f. https://conda-forge.org/blog/2020/10/29/macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock
* Ensure that the `HOST` compiler names that will be used at runtime are included (needed for cross-compilation) by replacing the `BUILD` values with `HOST` values in the compiler basenames in mg5_configuration.txt.

**MNT: Add procps-ng for linux for pgrep**

* Add `procps-ng` as a run requirement for linux platforms to provide pgrep for use in `madgraph/various/cluster.py`.
    - `procps-ng` is not available for `osx` platforms.
    - c.f. https://github.com/conda-forge/procps-ng-feedstock/issues/15
    - `madgraph/various/cluster.py` is used in Pythia8 related code.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

matthewfeickert commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13423230717](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13423230717). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13423235463](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13423235463). Examine the logs at this URL for more detail.</sub>

matthewfeickert commented:
@conda-forge-admin, please rerender

matthewfeickert committed:
FIX: Use target platform compiler names in mg5_configuration.txt

* Use CXX_FOR_BUILD and FF_FOR_BUILD to get the correct compilers that
  for the runtime target platform (the host platform). The '_FOR_BUILD'
  compilers give the build platform compilers, but with the correct name
  prefaced by the 'BUILD' triplet (e.g. x86_64-conda-linux-gnu).
   - c.f. https://conda-forge.org/blog/2020/10/29/macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock
* Ensure that the 'HOST' compiler names that will be used at runtime are
  included (needed for cross-compilation) by replacing the 'BUILD' values with
  'HOST' values in the compiler basenames in mg5_configuration.txt.

matthewfeickert commented:
As this is working, this should be packaged instead: https://github.com/conda-forge/staged-recipes/pull/31115

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

matthewfeickert commented:
Asked about `pgrep` for `osx-` platforms in https://github.com/conda-forge/procps-ng-feedstock/issues/15.

matthewfeickert committed:
FIX: Use mg5amcnlo-pythia8-interface for MG5aMC_PY8_interface

* Add 'mg5amcnlo-pythia8-interface' as a 'run' requirement, which supplies the
  Pythia8 interface program and defines pythia8 'run' requirements.
   - Skip 'mg5amcnlo-pythia8-interface' for 'osx-64' given PY8_parallelization splitting bug.
* Remove build information for MG5aMC_PY8_interface and remove file checks.
* Bump build number.

matthewfeickert committed:
MNT: Add procps-ng for linux for pgrep

* Add 'procps-ng' as a run requirement for linux platforms to provide pgrep for
  use in madgraph/various/cluster.py.
    - procps-ng is not available for osx platforms.
    - madgraph/various/cluster.py is used in Pythia8 related code.

matthewfeickert committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.09.27.05.44.00

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

matthewfeickert committed:
TST: Skip NLO tests for osx-64

* Given that there are issues with osx-64 NLO calculations with
  clang at the moment, don't run tutorial_amc-at-nlo.mg5 on
  osx-64.
